### ISSUE TITLE
Worse performance than ROIPooling and ROIAlign on VOC07

### ISSUE THREAD

**Issue by bowenc0221** [vacancy/PreciseRoIPooling#1]
Hi,

Thanks for sharing your work. I tried your code by directly replacing ROIPooling in [faster-rcnn.pytorch](https://github.com/jwyang/faster-rcnn.pytorch) using the provided example.
```
from prroi_pool import PrRoIPool2D

avg_pool = PrRoIPool2D(window_height, window_width, spatial_scale)
roi_features = avg_pool(features, rois)
```
I trained the Faster RCNN with ResNet-101 backbone on VOC2007 trainval set and evaluated the performance on the VOC2007 test set.

I got these results:

Pooling method    | #GPUs | batch size | lr        | lr_decay | max_epoch     |  mAP
---------|--------|-----|--------|-----|-----|-----
ROI Pooling    | 4 | 16 | 0.016 | 5   | 7   | 74.3
ROI Align    | 4 | 16 | 0.016 | 5   | 7   | 75.2
Precise ROI Pooling    | 4 | 16 | 0.016 | 5   | 7   | 73.8

I'm using python 2.7 so I modified [this code](https://github.com/vacancy/PreciseRoIPooling/blob/master/pytorch/prroi_pool/travis.sh#L17) to `python build.py` and [this code](https://github.com/vacancy/PreciseRoIPooling/blob/master/pytorch/prroi_pool/prroi_pool.py#L21) to `super(PrRoIPool2D, self).__init__()`.

I haven't read [faster-rcnn.pytorch](https://github.com/jwyang/faster-rcnn.pytorch) codes carefully, but it seems that they followed the caffe implementation to use `(batch_index, x0, y0, x1, y1)` for rois.

Do you have any idea what might cause this degraded performance on VOC07 dataset? Do I need to change training parameters for Precise ROIPooling?

**Comment by YuningJiang** [comment]
I guess it is due to the released PrROI Pooling is an AVERAGE version, while the ROI Align uses MAX instead. You can try to use ROI Align (AVERAGE) for a comparison.

**Comment by xmyqsh** [comment]
interesting

**Comment by ZNorah** [comment]
I also tried it, and got these results:

Pooling method | #GPUs | batch size | lr | lr_decay | max_epoch | mAP
-- | -- | -- | -- | -- | -- | --
ROI Pooling | 1 | 1 | 1e-3 | 5 | 7 | 74.6
ROI Align(AVG) | 1 | 1 | 1e-3 | 5 | 7 | 74.9
Precise ROI Pooling | 1 | 1 | 1e-3 | 5 | 7 | 75.3
